cmd/capibmadm/cmd/powervs/key/create.go

```diff
@@ -0,0 +1,84 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package key
+
+import (
+	"context"
+
+	"github.com/spf13/cobra"
+
+	"github.com/IBM-Cloud/power-go-client/clients/instance"
+	"github.com/IBM-Cloud/power-go-client/power/models"
+
+	"sigs.k8s.io/cluster-api/cmd/clusterctl/log"
+
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/iam"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/clients/powervs"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/options"
+	"sigs.k8s.io/cluster-api-provider-ibmcloud/cmd/capibmadm/utils"
+)
+
+type keyCreateOptions struct {
+	key     string
+	keyName string
+}
+
+// CreateSSHKeyCommand - child command of 'key' to create an SSH key.
+func CreateSSHKeyCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create",
+		Short: "Create an SSH key in the PowerVS environment.",
+		Example: `
+# Create an SSH key.
+export IBMCLOUD_API_KEY=<api-key>
+capibmadm powervs key create --name <key-name> --key "<ssh-key>" --service-instance-id <service-instance-id> --zone <zone>`,
+	}
+
+	var keyCreateOption keyCreateOptions
+	cmd.Flags().StringVar(&keyCreateOption.key, "key", "", "SSH RSA key string within a double quotation marks. For example, \"ssh-rsa AAA... \".")
```

Comment:
Most of the folks usually prefer file over a plain text of a key for better UX, wondering if can enhance the code to do so. One more flag can be added like `key-file` which reads from the file and does some validation as well.

@Prajyot-Parab @Karthik-K-N wdys?

>Note: if agree - we can have a follow up PR

Reply:
sure we can introduce a flag , Once this get merged, may be @kishen-v do you want to give a try for that.

Reply:
@Karthik-K-N, anything works. If all the PRs for capibmdm needs to go together, I'll work on an enhancement PR to support reading of SSH key from file.

Reply:
+1 for `--key-file` flag

Reply:
Changes are made support the `--key-file` argument.
```
kishenv@Kishens-MacBook-Pro capibmadm % ./capibmadm powervs key create --name capiadm-sshkey2 --key-path /Users/kishenv/.ssh/id_rsa.pub --zone osa21 --service-instance-id <redacted>
Creating SSH key...
Successfully created the SSH key. name="capiadm-sshkey2"